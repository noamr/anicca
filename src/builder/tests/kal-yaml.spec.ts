import {parse} from 'yaml'
const parseKal = require('../parser.ts')

const kalYaml = `
View myView:
    '#output':
        content: value
    '#increment':
        on click:
        - dispatch increment to myController
        - run script:
            e => e.preventDefault()
Controller myController:
    state root:
        entering:
            value = 0

        on increment:
            value += 1

Let value: u32
`

describe('kal yaml parsing', () => {
    it('should be valid yaml', () => {
        expect(parse(kalYaml)).toMatchSnapshot()
    })
    it('should be parsed to json', () => {
        expect(parseKal(parse(kalYaml))).toMatchSnapshot()
    })
})