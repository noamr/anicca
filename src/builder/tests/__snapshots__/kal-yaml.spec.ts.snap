// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kal yaml parsing should be parsed to json 1`] = `
Array [
  Object {
    "name": "myView",
    "rules": Array [
      Object {
        "declarations": Array [
          Object {
            "src": Object {
              "$ref": "value",
              "$token": Object {
                "col": 1,
                "line": 1,
              },
            },
            "targetType": "content",
            "type": "Bind",
          },
        ],
        "selector": "#output",
      },
      Object {
        "declarations": Array [
          Object {
            "actions": Array [
              Object {
                "event": "increment",
                "target": "myController",
                "type": "Dispatch",
              },
              Object {
                "source": "e => e.preventDefault()",
                "type": "RunScript",
              },
            ],
            "eventType": "click",
            "type": "DOMEvent",
          },
        ],
        "selector": "#increment",
      },
    ],
    "type": "View",
  },
  Object {
    "name": "myController",
    "rootState": Object {
      "children": Array [
        Object {
          "actions": Array [
            Object {
              "src": Object {
                "$primitive": 0,
              },
              "target": Object {
                "$ref": "value",
              },
              "type": "Assign",
            },
          ],
          "type": "OnEntry",
        },
        Object {
          "actions": Array [
            Object {
              "src": Object {
                "args": Array [
                  Object {
                    "$ref": "value",
                  },
                  Object {
                    "$primitive": 1,
                  },
                ],
                "op": "add",
              },
              "target": Object {
                "$ref": "value",
              },
              "type": "Assign",
            },
          ],
          "event": "increment",
          "type": "Transition",
        },
      ],
      "name": "root",
      "type": "State",
    },
    "type": "Controller",
  },
  Object {
    "name": "value",
    "type": "Let",
    "value": "u32",
  },
]
`;

exports[`kal yaml parsing should be valid yaml 1`] = `
Object {
  "Controller myController": Object {
    "state root": Object {
      "entering": "value = 0",
      "on increment": "value += 1",
    },
  },
  "Let value": "u32",
  "View myView": Object {
    "#increment": Object {
      "on click": Array [
        "dispatch increment to myController",
        Object {
          "run script": "e => e.preventDefault()",
        },
      ],
    },
    "#output": Object {
      "content": "value",
    },
  },
}
`;
